src/log/log.c

```diff
@@ -1517,17 +1520,20 @@ __log_truncate_file(WT_SESSION_IMPL *session, WT_FH *log_fh, wt_off_t offset)
 	WT_CONNECTION_IMPL *conn;
 	WT_DECL_RET;
 	WT_LOG *log;
+	bool truncate;
 
 	conn = S2C(session);
 	log = conn->log;
+	truncate = false;
 
 	if (!F_ISSET(log, WT_LOG_TRUNCATE_NOTSUP) && !conn->hot_backup) {
-		__wt_readlock(session, &conn->hot_backup_lock);
-		if (conn->hot_backup)
-			__wt_readunlock(session, &conn->hot_backup_lock);
-		else {
-			ret = __wt_ftruncate(session, log_fh, offset);
-			__wt_readunlock(session, &conn->hot_backup_lock);
+		WT_WITH_HOTBACKUP_LOCK(session, {
+			if (!conn->hot_backup) {
+				truncate = true;
+				ret = __wt_ftruncate(session, log_fh, offset);
+			}
+		});
+		if (truncate) {
```

Comment:
There were a few cases like this where I couldn't easily reproduce the sequence of locks/unlocks. So I had to introduce a variable to know whether to continue with the `else` block after I've released the lock.